factory/usr/lib/systemd/system/systemd-bootchart.service

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Boot Process Profiler
+Documentation=man:systemd-bootchart.service(1) man:bootchart.conf(5)
+DefaultDependencies=no
+
+[Service]
+ExecStartPre=/usr/sbin/mkdir -p /run/log
+ExecStart=/lib/systemd/systemd-bootchart -r
+KillSignal=SIGHUP
```

Comment:
why is this here? I don't see it in the normal systemd-bootchart service on classic:

```
$ systemctl cat systemd-bootchart
# /lib/systemd/system/systemd-bootchart.service
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

# Note: it's usually a better idea to run systemd-bootchart via the
# init= kernel command line switch. See the man page for details.

[Unit]
Description=Boot Process Profiler
Documentation=man:systemd-bootchart.service(1) man:bootchart.conf(5)
DefaultDependencies=no

[Service]
ExecStart=/lib/systemd/systemd-bootchart -r

[Install]
WantedBy=sysinit.target
```

Reply:
If killed by SIGTERM, `systemd-bootchart` does not generate the bootchart (note that the bootchart file is written right when the daemon exits). SIGHUP is handled appropriately to make sure the file is actually written. This is arguably and upstream bug, imo SIGTERM and SIGNINT should be handled in the same way as SIGHUP. But, for the moment we can specify `KillSignal` to avoid applying patches to `systemd-bootchart`.

Reply:
thanks for the explanation